pkg/apis/machine/v1beta1/machine_webhook.go

```diff
@@ -144,9 +162,10 @@ func getInfra() (*osconfigv1.Infrastructure, error) {
 	return infra, nil
 }
 
-type machineAdmissionFn func(m *Machine, clusterID string) (bool, utilerrors.Aggregate)
+type machineAdmissionFn func(m *Machine, clusterID string, client *client.Client) (bool, utilerrors.Aggregate)
```

Comment:
Why use a pointer here? `client.Client` is an interface, I wouldn't have expected a pointer to an interface to be needed normally